eq-author/src/App/QuestionnaireDesignPage/index.js

```diff
@@ -81,31 +81,48 @@ export class UnwrappedQuestionnaireDesignPage extends Component {
     } = this.props;
     const { entityName, entityId } = match.params;
 
-    let sectionId, position;
-    if (entityName === SECTION) {
-      sectionId = entityId;
-      position = 0;
-    } else if (entityName === PAGE) {
-      for (let i = 0; i < questionnaire.sections.length; ++i) {
-        const section = questionnaire.sections[i];
-        const page = find(section.pages, { id: entityId });
-        if (page) {
-          sectionId = section.id;
-          position = section.pages.indexOf(page) + 1;
-          break;
+    let sectionId, position, selectedPage, selectedSection, selectedFolder;
+
+    switch (entityName) {
+      case SECTION:
+        sectionId = entityId;
+        position = 0;
+        break;
+
+      case QUESTION_CONFIRMATION:
+      // Fall through
+      case PAGE:
+        for (const section of questionnaire.sections) {
+          for (const folder of section.folders) {
+            for (const page of folder.pages) {
+              const comparatorID =
+                entityName === QUESTION_CONFIRMATION
+                  ? page.confirmation.id
+                  : page.id;
+              if (comparatorID === entityId) {
+                selectedPage = page;
+                selectedSection = section;
+                selectedFolder = folder;
+                break;
+              }
+            }
+          }
         }
-      }
-    } else if (entityName === QUESTION_CONFIRMATION) {
-      for (let i = 0; i < questionnaire.sections.length; ++i) {
-        const section = questionnaire.sections[i];
-        const page = find(section.pages, { confirmation: { id: entityId } });
-        if (page) {
-          sectionId = section.id;
-          position = section.pages.indexOf(page) + 1;
-          break;
+
+        if (selectedPage) {
+          sectionId = selectedSection.id;
+          position = selectedSection.folders.indexOf(selectedFolder) + 1;
         }
-      }
+
+        break;
+
+      // case FOLDER:
+      // TODO
+
+      default:
+        throw new Error("Unrecognised entity type.");
```

Comment:
Entity name or type?

Reply:
Could be `name` instead if you want. That line should never really get hit in practice - was just needed since ESLint insists on a `default` case (although it might help in dev in the rare case we mess up the name for a new entity type)

Reply:
Just for the sake of future clarity. I'm not particularly bothered but someone else might be

Reply:
Okay, no probs. I'll rename